pack.ps1

```diff
@@ -0,0 +1,146 @@
+		Write-Verbose "Preparing file: $($dllPath)"
+        Copy-Item -Path $dllPath -Destination $libDir
+		if (-not $version) {
+            $version = [System.Reflection.AssemblyName]::GetAssemblyName($dllPath).Version.ToString()
+		}
+		
+        if ($IncludePdbs) {
+            $dllPath = Join-Path -Path $frameworkPathItem.FullName -ChildPath "Microsoft.PackageManagement.NuGetProvider.pdb"
+			Write-Verbose "Preparing file: $($dllPath)"
+			Copy-Item -Path $dllPath -Destination $libDir
+        }
+    }
+
+    if (-not $version) {
+        throw "Couldn't discover version of package - is Microsoft.PackageManagement.dll there?"
+    }
+
+    if ($PrereleaseString) {
+        $version += "-$PrereleaseString"
+    }
+
+    $nuspecContents = Get-Content -Path (Join-Path -Path $PSScriptRoot -ChildPath 'NuGetProvider.nuspec') | Out-String
+    Write-Verbose "Setting Package.Id = $PackageId"
+    $nuspecContents = $nuspecContents -replace "{name}",$PackageId
+    Write-Verbose "Setting Package.Version = $version"
+    $nuspecContents = $nuspecContents -replace "{version}",$version
+    Write-Verbose "Setting Package.Authors = $Authors"
+    $nuspecContents = $nuspecContents -replace "{authors}",$Authors
+    Write-Verbose "Setting Package.Owners = $Owners"
+    $nuspecContents = $nuspecContents -replace "{owners}",$Owners
+    if ($ReleaseNotesFile) {
+        if (-not (Test-Path -Path $ReleaseNotesFile)) {
+            throw "Release notes file '$ReleaseNotesFile' does not exist"
+        }
+
+        $ReleaseNotes = Get-Content -Path $ReleaseNotesFile | Out-String
+    } else {
+        $ReleaseNotes = $ReleaseNotes | Out-String
+    }
+
+    Write-Verbose "Setting Package.ReleaseNotes = '$ReleaseNotes'"
+    $nuspecContents = $nuspecContents -replace "{releaseNotes}",$ReleaseNotes
+    $nuspecName = "temp.nuspec"
+    $nuspecContents | Out-File -FilePath (Join-Path -Path $tempDir -ChildPath $nuspecName)
+    Write-Verbose "Packing..."
+    Push-Location -Path $tempDir
+    & $nugetPath pack $nuspecName
+    Pop-Location
+    if ($LastExitCode -gt 0) {
+        throw 'NuGet.exe pack failed. See previous error messages.'
+    }
+
+    Write-Verbose "Copying package to $PackageOutDir"
+    Get-ChildItem -Path (Join-Path -Path $tempDir -ChildPath "*.nupkg") | Select-Object -First 1 | Select-Object -ExpandProperty FullName | Copy-Item -Destination $PackageOutDir
+} finally {
+    $tries = 3
+    while ((Test-Path -Path $tempDir) -and ($tries -gt 0)) {
+        try {
+            $null = Remove-Item -Path $tempDir -Recurse -Force -ErrorAction Ignore
+        } catch {
+        }
+
+        if ((Test-Path -Path $tempDir) -and ($tries -gt 0)) {
+            Start-Sleep -Milliseconds (100 * $tries)
+            $tries = $tries - 1
+        }
+    }
+
+    if (Test-Path -path $tempDir) {
+        Write-Error "Failed to remove temp directory: $tempDir"
```

Comment:
Whether we can delete tempdir or not won't impact our packing, right? If so I recommend write-warning.

Reply:
2592ee5